### PR TITLE
#66, #67, #3, #42, #39, #5, #71, #72, #73, #77, #62, #63

### PR DESCRIPTION
**Pull request by kentmw**
… view, and no render until attach

### CODE REVIEW

File: modules/FormModel.js [hunk 1]
```diff
@@ -696,21 +689,21 @@
...
             _.each(modelConfig.fields, function(field) {
-              args.push(formModel._clone(modelConfig.model.get(field)));
+              args.push(formModel.__cloneVal(modelConfig.model.get(field)));
             });
           } else {
-            args.push(formModel._clone(modelConfig.model.attributes));
+            args.push(formModel.__cloneVal(modelConfig.model.attributes));
           }
         });
         pullCallback.apply(formModel, args);
```
Comment (arikwex) [inline modules/FormModel.js]: What's our test coverage look like for this file?

File: modules/ListView.js [hunk 2]
```diff
@@ -107,21 +107,21 @@
...
-    _modelName: '',
-    _childView: null,
-    _modelToViewMap: null,
-    _template: null,
-    _emptyTemplate: null,
-    _childContext: null,
-    _renderWait: 0,
-    _delayedRender: null,
-    _delayedRenderTimeout: null,
+    collection: null,
```
Comment (arikwex) [inline modules/ListView.js]: Do these properties need comments?

File: modules/View.js [hunk 3]
```diff
@@ -20,12 +20,26 @@
...
    */
   var View = Backbone.View.extend({
-    _childViews: null,
     viewState: null,
     template: null,
-    _isActive: false,
-    _isAttached: false,
-    _isDisposed: false,
+    feedback: null,
+    feedbackModel: null, // TODO feedback model shouldn't be called a model if it's a cell
```
Comment (arikwex) [inline modules/View.js]: Create issue and remove TODO

File: modules/View.js [hunk 4]
```diff
@@ -352,8 +389,337 @@
...
+      this.feedbackModel.trigger('change:' + feedbackModelField);
+    },
+
+    /**
+     * Creates the "when" bindings, and collates and invokes the "then" methods for all feedbacks
+     * Finds all feedback zones that match the "to" field, and binds the "when" events to invoke the "then" method
+     * @private
+     * @method __generateFeedbackBindings
+     */
+    __generateFeedbackBindings: function() {
```
Comment (arikwex) [inline modules/View.js]: This method is long.  I know that it's a port from FormView... but it's still too long.
Let's refactor this (not in this commit) to be a bit more readable.

File: package.json [hunk 5]
```diff
@@ -51,13 +52,14 @@
...
+    "node-dir": "0.1.8 ",
+    "path": "0.11.14",
     "promise": "6.1.0",
     "require-dir": "^0.1.0",
     "requireify": "^0.2.1",
     "vinyl-buffer": "^1.0.0",
     "vinyl-source-stream": "^1.0.0",
-    "watchify": "^2.4.0",
-    "handlebars": "3.0.0"
+    "watchify": "^2.4.0"
```
Comment (arikwex) [inline package.json]: uhm.  Does this actually use watchify?  Or is this quick-sip legacy?

File: test/spec/functional/listView.js [hunk 6]
```diff
@@ -237,28 +241,28 @@ describe('A List View', function() {
...
-    expect(_.last(myListView.$el.children())).toBe(myListView.getChildView(model).el);
+    expect(_.first(myListView.$el.children())).toBe(myListView.getChildViewFromModel(model3).el);
+    expect(_.last(myListView.$el.children())).toBe(myListView.getChildViewFromModel(model).el);
   });
 
   it('can sort a large list in a reasonable time', function() {
     var startTime, endTime, i,
         numberOfViews = 1000,
-        threshold = 300;
+        threshold = 500;
```
Comment (arikwex) [inline test/spec/functional/listView.js]: This test is always awkward...  Ideally it should be machine agnostic...  We might need to think of a more cross-cpu-friendly way of conducting this test.

File: modules/FormModel.js (hunk 1 above)
Comment (kentmw) [inline modules/FormModel.js]: Really good. FormModel is one of the most tested modules

File: package.json (hunk 5 above)
Comment (kentmw) [inline package.json]: Yes, @mandragorn added watchify for tests

File: modules/ListView.js (hunk 2 above)
Comment (kentmw) [inline modules/ListView.js]: done

File: modules/View.js (hunk 3 above)
Comment (kentmw) [inline modules/View.js]: done

File: modules/collectionLoadingMixin.js [hunk 7]
```diff
@@ -20,21 +20,16 @@
...
-        loading = options.loading;
+    loadingMixin = function(collection) {
 
       /**
        * @method hasLoadedOnce
        * @return true if this collection has ever loaded from a fetch call
        */
       collection.hasLoadedOnce = function() {
-        return loadedOnce;
+        return collection.loadedOnce;
```
Comment (arikwex) [inline modules/collectionLoadingMixin.js]: use "this"

File: modules/collectionLoadingMixin.js [hunk 8]
```diff
@@ -59,30 +54,30 @@
...
       /**
        * Base load function that will trigger a "load-begin" and a "load-complete" as
        * the fetch happens. Use this method to wrap any method that returns a promise in loading events
-       * @method _loadWrapper
+       * @method __loadWrapper
        * @param fetchMethod {Function} - the method to invoke a fetch
        * @param options {Object} - the object to hold the options needed by the fetchMethod
        * @return a promise when the fetch method has completed and the events have been triggered
        */
-      collection._loadWrapper = function(fetchMethod, options) {
```
Comment (arikwex) [inline modules/collectionLoadingMixin.js]: future: try adding this mixin to models?

File: modules/collectionLoadingMixin.js [hunk 9]
```diff
@@ -90,16 +85,15 @@
     return {
```
Comment (arikwex) [inline modules/collectionLoadingMixin.js]: dont use the module pattern anymore.  just put this in the return hash

File: modules/collectionRegistrationMixin.js [hunk 10]
```diff
@@ -357,42 +340,49 @@
...
+          this.knownPrivateCollections = {};
+          this.getByIdsUrl = options.getByIdsUrl || '/ids';
+          this.fetchHttpAction = 'POST';
+          this.lazyFetch = options.lazyFetch || false;
+          this.fetchUsingTrackedIds = options.fetchUsingTrackedIds !== false;
+          cacheMixin(this);
+        } else {
+          this.trackedIds = [];
+          this.listenTo(this.parentInstance, 'load-begin', function() {
+            this.trigger('cache-load-begin');
```
Comment (arikwex) [inline modules/collectionRegistrationMixin.js]: does "this" still mean what you think it does? and what about payload

File: modules/collectionLoadingMixin.js (hunk 7 above)
Comment (kentmw) [inline modules/collectionLoadingMixin.js]: done

File: modules/collectionLoadingMixin.js (hunk 9 above)
Comment (kentmw) [inline modules/collectionLoadingMixin.js]: done

File: modules/collectionRegistrationMixin.js (hunk 10 above)
Comment (kentmw) [inline modules/collectionRegistrationMixin.js]: done and done. "this" is correct, this is under test, the payload we can add if someone wants something.
the order goes: load-begin->cache-load-begin->FETCH->cache-load-complete->load-complete

File: modules/collectionLoadingMixin.js (hunk 8 above)
Comment (kentmw) [inline modules/collectionLoadingMixin.js]: Added #78

**Comment by arikwex** [comment]
+2

**Pull request merged**